lib/migration-stream.js

```diff
@@ -29,32 +32,70 @@ function MigrationStream() {
     });
 
     if (v2 === true) {
-      // Check for errors on the v2 tile
-      var err = checkForErrors(info);
-      if (err) return callback(err);
-
-      migrationStream.push(tile);
-      return callback();
-    } else {
+      decompressBuffer(tile.buffer, function(err, decompressedBuffer){
+          if (err) {
+            err.code = 'EINVALID';
+            return callback(err);
+          } 
+
+          vtvalidate.isValid(decompressedBuffer, function(err, result){
+            // returns empty string if it's a valid tile
+            // if invalid tile - string that specifies why the tile is invalid
+            if (result === 'ClosePath command count is not 1' || result === 'Max count too large'){
+              var error = new Error(result);
+              error.code = 'EINVALID';
+              return callback(error);
+            
+            }else if(result){
+
+              var newValidatorErrors = result; 
```

Comment:
best practice to declare new vars within the scope that they're used. So for `newValidatorErrors`, since we're using it outside of this `else if` block, it should be created up above.

Reply:
fixed!